### PR TITLE
Sort category and non-category nav link according to order in left menu

### PR DESCRIPTION
**Pull request by yuye-aws**
### Description

Sort category and non-category nav link according to order in left menu.
Please note that the value of order for **Home** and **Overview** has been changed.

## Screenshot

image

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/core/public/chrome/ui/header/collapsible_nav.tsx [hunk 1]
```diff
@@ -71,6 +71,45 @@ function getOrderedCategories(
   );
 }
 
+function getMergedNavLinks(
+  unknowns: ChromeNavLink[],
```
Comment (SuZhou-Joe) [inline src/core/public/chrome/ui/header/collapsible_nav.tsx:75]: Are these `unknowns` sorted array?

File: src/core/public/chrome/ui/header/collapsible_nav.tsx [hunk 2]
```diff
@@ -71,6 +71,45 @@ function getOrderedCategories(
...
+  orderedCategories: string[],
+  categoryDictionary: ReturnType<typeof getAllCategories>
+): Array<string | ChromeNavLink> {
+  unknowns = sortBy(unknowns, 'order');
```
Comment (SuZhou-Joe) [inline src/core/public/chrome/ui/header/collapsible_nav.tsx:79]: It is not a good practice to modify the original parameters inside a function call, can we create a new temp variable to store the sorted unknowns links?

**Review by SuZhou-Joe — APPROVED** [review]

File: src/core/public/chrome/ui/header/collapsible_nav.tsx [hunk 3]
```diff
@@ -71,6 +71,48 @@ function getOrderedCategories(
   );
 }
 
+function getMergedNavLinks(
```
Comment (wanglam) [inline src/core/public/chrome/ui/header/collapsible_nav.tsx:74]: This `getMergedNavLinks` function is a little bit hard to understand. Could we refactor it based below logic:
- separate `unknowns` to two groups, one for has order property, one for not
- combine `orderedCategories` and `categoryDictionary` to one array with category string and order, than separate two groups like `unknowns`
- combine `unknowns` and `new array` with order property in one array, than sort by order
- append all `unknowns` without order property
- append all `categories` without order property

File: src/core/public/chrome/ui/header/collapsible_nav.tsx [hunk 4]
```diff
@@ -71,6 +71,29 @@ function getOrderedCategories(
...
+    .filter((categoryName) => categoryDictionary[categoryName]?.order !== null)
+    .map((categoryName) => ({ categoryName, order: categoryDictionary[categoryName]?.order }));
+  const orderedCategoryWithoutOrder = orderedCategories.filter(
+    (categoryName) => categoryDictionary[categoryName]?.order === null
+  );
+  const mergedNavLinks = sortBy(
+    [...uncategorizedLinksWithOrder, ...orderedCategoryWithOrder],
+    'order'
+  ).map((navLink) => ('categoryName' in navLink ? navLink.categoryName : navLink));
+  return [...mergedNavLinks, ...orderedCategoryWithoutOrder, ...uncategorizedLinksWithoutOrder];
```
Comment (SuZhou-Joe) [inline src/core/public/chrome/ui/header/collapsible_nav.tsx:94]: All LGTM, but here we'd better add a comment to say that we sort links with category before those without category by `convention` that categoried links appear before uncategoried ones.

**Review by wanglam — APPROVED** [review]

**Pull request merged**